pulsar-client-cpp/lib/PartitionedProducerImpl.cc

```diff
@@ -379,8 +418,8 @@ bool PartitionedProducerImpl::isConnected() const {
     Lock producersLock(producersMutex_);
     const auto producers = producers_;
     producersLock.unlock();
-    for (const auto& producer : producers_) {
-        if (!producer->isConnected()) {
+    for (const auto& producer : producers) {
```

Comment:
If the producer has been created, but no messages have been sent, this will return true, even though there has been no connections.

Reply:
Yes. The question is what people do with this method. Is it an indicator of a problem? Without lazy producers, false means we were once connected but now aren't and so can be considered a transient problem. But with lazy producers, false may mean we've not even tried, so false does not mean anything. So I decided to return true to signal absence of a problem.

Definitely interested to hear opinions on that.

Reply:
Ya, I have no strong opinion here. "connected" is a weird concept in a distributed system like this.